target/linux/ipq40xx/base-files/lib/upgrade/platform.sh

```diff
@@ -110,6 +110,11 @@ platform_do_upgrade() {
 		CI_KERNPART="part.safe"
 		nand_do_upgrade "$1"
 		;;
+	mikrotik,hap-ac2|\
+	mikrotik,sxtsq-5-ac)
+		[ "$(rootfs_type)" = "tmpfs" ] && mtd erase firmware
+		default_do_upgrade "$1"
+		;;
```

Comment:
Likewise here we will end up enumerating all devices instead of having a catchall. Maybe this could be mitigated with `mikrotik,*)`.
@robimarko

Reply:
Not that won't work for all of the devices like this only works for SPI-NOR based ones, there are also NAND based ones like hAP ac3 for which this does not work.

Honestly, I don't see an issue with this.

Reply:
You can special case the NAND devices before the catchall.
That being said, I will reiterate: we should NOT add support for more NAND devices. We are deceiving users and we will brick their devices in the end.

Reply:
Yeah, we can do that.
I personally prefer listing every device as it makes it really clear, especially with the MikroTik revision mess.

Why not?
IPQ40xx based NAND ones like RB450Gx4 are not like the MIPS ones.
These require the kernel to be packaged in UBIFS partition while rootfs uses the UBI wrapper, so bad blocks are actually managed normally.

Reply:
> Yeah, we can do that.
> I personally prefer listing every device as it makes it really clear, especially with the MikroTik revision mess.

It also wastes space and makes script execution slower...

> Why not?
> IPQ40xx based NAND ones like RB450Gx4 are not like the MIPS ones.
> These require the kernel to be packaged in UBIFS partition while rootfs uses the UBI wrapper, so bad blocks are actually managed normally.

That's good news. But then I don't understand why the NOR devices still use YAFFS if mikrotik is already switching to UBIFS on NAND. Surely they realized that YAFFS is going the way of the dinosaurs? Maybe it's possible to get rid of kernel2minor on the ipq devices eventually?

Reply:
> You can special case the NAND devices before the catchall.
> That being said, I will reiterate: we should NOT add support for more NAND devices. We are deceiving users and we will brick their devices in the end.

Maybe off-topic here, but with the next stable close, I would expect that the issues won't be fixed in time anymore.

I wonder whether we should put `DEFAULT := n` for the relevant Mikrotik devices (just NAND or even all of them). Opinions?